test/dense.jl

```diff
@@ -15,3 +15,71 @@ end
         (sum ∘ dense)(x)
     end
 end
+
+@testset "Dense grad" begin
+    # #########
+    # # utils #
+    # #########
+
+    function get_model_and_params(layers)
+        model = identity
+        ps = []
+        for layer in layers
+            model = layer ∘ model
+            for param in Scapa.params(layer)
+                push!(ps, param)
+            end
+        end
+
+        return model, ps
+    end
```

Comment:
I think with this design, it would be hard to define function with branches.

Reply:
Yes, ur right XD